stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java

```diff
@@ -174,6 +176,82 @@ public void enumTest() throws Exception {
+    filter =
+        new FlightFilter()
+            .setFilterInputBooleanOperationParameter(
+                (f) ->
+                    createOr(
+                        f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int1),
+                        f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int2)));
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 16", flightList, Arrays.asList("1", "2", "3", "4"));
+
+    // Case 17: filter flight on a boolean clause (AND)
+    filter =
+        new FlightFilter()
+            .setFilterInputBooleanOperationParameter(
+                (f) ->
+                    createAnd(
+                        f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int1),
+                        f.makeInputPredicate("in1", FlightFilterOp.EQUAL, string1)));
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 17", flightList, Arrays.asList("1"));
+
+    // Case 18: filter flight with nested a boolean clauses
+    filter =
+        new FlightFilter()
+            .setFilterInputBooleanOperationParameter(
+                (f) ->
+                    createOr(
+                        createAnd(
+                            f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int1),
+                            f.makeInputPredicate("in1", FlightFilterOp.EQUAL, string1)),
+                        createAnd(
+                            f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int2),
+                            f.makeInputPredicate("in1", FlightFilterOp.EQUAL, string2))));
+
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 18", flightList, Arrays.asList("1", "2"));
+
+    // Case 19: filter flight with a boolean clause and an in clause
+    filter =
+        new FlightFilter()
+            .setFilterInputBooleanOperationParameter(
+                (f) ->
+                    createAnd(
+                        f.makeInputPredicate("in0", FlightFilterOp.EQUAL, int1),
+                        f.makeInputPredicate(
+                            "in1", FlightFilterOp.IN, Arrays.asList(string1, string2))));
+
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 19", flightList, Arrays.asList("1", "3"));
+
+    // Case 20: filter flight with a null check for a field that doesn't exist
+    filter = new FlightFilter().addFilterInputParameter("in10000", FlightFilterOp.EQUAL, null);
+
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 20", flightList, Arrays.asList("0", "1", "2", "3", "4", "5"));
+
+    // Case 21: filter input on an in clause with a POJO
+    filter =
+        new FlightFilter()
+            .addFilterInputParameter("in2", FlightFilterOp.IN, Arrays.asList(pojo1, pojo2));
+    flightList = flightDao.getFlights(0, 100, filter);
+    checkResults("case 21", flightList, Arrays.asList("1", "2", "3", "4"));
```

Comment:
It looks like this code supports using boolean operations and other operations (like `addFilterInputParameter()`)  in the same filter. If so, is this functionality tested? I didn't see a test for it here.

Reply:
Good eye.  I'll add a test for that